script.js

```diff
@@ -1,4 +1,147 @@
+  }
+  //PLAYER 1 STARTS GAME//
+  if (gameState === "start" && input === "") {
+    gameState = "player1";
+    output.innerHTML = "Enter";
+    return playerStartGame(
+      diceCombinedPlayerOneFirst,
+      diceCombinedPlayerOneSecond,
+      "1"
+    );
+  }
+  //PLAYER 1 TYPES '1' AS DICE OPTION 1//
+  if (gameState === "player1") {
+    if (input === "1") {
+      output.innerHTML = "Roll";
+      return getResultsPlayerOne(diceCombinedPlayerOneFirst);
+    }
+    //PLAYER 1 TYPES '2' AS DICE OPTION 2//
+    if (input === "2") {
+      output.innerHTML = "Roll";
+      return getResultsPlayerOne(diceCombinedPlayerOneSecond);
+    }
+    //PLAYER 2 STARTS GAME//
+    if (input === "" && output.innerHTML !== "Enter") {
+      gameState = "player2";
+      output.innerHTML = "Enter";
+      return playerStartGame(
+        diceCombinedPlayerTwoFirst,
+        diceCombinedPlayerTwoSecond,
+        "2"
+      );
+    } else {
+      return "Error, please only key '1' or '2' to choose your dice order.";
+    }
+  }
+
+  //PLAYER 2 TYPES '1' AS DICE OPTION 1//
+  if (gameState === "player2") {
+    if (input === "1") {
+      output.innerHTML = "Restart";
+      return getResultsPlayerTwo(diceCombinedPlayerTwoFirst);
+    }
+    //PLAYER 2 TYPES '2' AS DICE OPTION 2//
+    if (input === "2") {
+      output.innerHTML = "Restart";
+      return getResultsPlayerTwo(diceCombinedPlayerTwoSecond);
+    }
+    //GAME RESTARTS//
+    if (input === "" && output.innerHTML != "Enter") {
+      gameState = "start";
+      reset();
+
+      output.innerHTML = "Roll";
+      return "Welcome to the game of Beat That!<br><br>This is a 2-player game. You will roll 2 dice, and you can choose the order of the dice to form the largest number. The player with the largest number wins!<br><br>🎲 Player 1, please roll the dice. 🎲";
+    } else {
+      return "Error, please only key '1' or '2' to choose your dice order.";
+    }
+  }
+
+  //INVALID INPUT//
+  if ((gameState === 0 || gameState === start) && input != "") {
+    return "Error, please roll the dice.";
+  }
+};
+
+//==============HELPER FUNCTION=================//
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+var getResultsPlayerTwo = function (diceResult) {
+  playerTwoFinalResult = diceResult.join("");
+  var winner;
+  if (playerOneFinalResult > playerTwoFinalResult) {
+    winner = "Player 1!";
+  } else if (playerTwoFinalResult > playerOneFinalResult) {
+    winner = "Player 2!";
+  } else {
+    winner = "It's a draw!";
+  }
+  return (
+    "Player 2, your combined number is: " +
+    diceResult.join("") +
+    ".<br><br>Player 1 highest number: " +
+    playerOneFinalResult +
+    "<br>Player 2 highest number: " +
+    playerTwoFinalResult +
+    "<br>Winner: " +
+    winner +
+    "<br><br>Click 'restart' to play again."
+  );
```

Comment:
would like to see you abstract this into another function, and you can essentially turn `getResultsPlayerTwo` and `getResultsPlayerOne` into a singular reusable function. It would make more sense as well by reading the name. I wouldnt expect the winning condition to happen out of a function named `getResultsPlayerTwo`